play-akka-http-ws-standalone/src/main/java/play/libs/ws/akkahttp/StandaloneAkkaHttpWSRequest.java

```diff
@@ -678,6 +689,38 @@ private StandaloneWSRequest copy(Duration timeout) {
     return new StandaloneAkkaHttpWSRequest(this.request, this.filters, timeout, this.sys, this.mat, this.ctx, this.config);
   }
 
+  private StandaloneWSRequest copy(WSClientConfig config) {
+    return new StandaloneAkkaHttpWSRequest(this.request, this.filters, this.timeout, this.sys, this.mat, this.ctx, config);
+  }
+
+  private CompletionStage<HttpResponse> requestToResponse(HttpRequest request) {
+    return Http.get(sys).singleRequest(request, ctx);
+  }
+
+  private CompletableFuture<HttpResponse> handleRedirects(Integer redirectsLeft, HttpRequest request, HttpResponse response) {
+    if (redirectsLeft < 1) {
+      final CompletableFuture<HttpResponse> exception = new CompletableFuture<>();
+      exception.completeExceptionally(new TimeoutException("Maximum redirect reached: " + MAX_REDIRECTS));
+      return exception;
+    } else if (response.status().isRedirection() && config.followRedirects()) {
+      final Uri location = response.getHeader(Location.class).get().getUri();
+      Uri redirectUri = location;
+      if (location.isRelative()) {
+        if (location.path().startsWith("/")) {
+          redirectUri = request.getUri().path(location.path());
+        }
+        else {
+          redirectUri = request.getUri().path(request.getUri().path() + location.path());
+        }
+      }
+      return requestToResponse(request.withUri(redirectUri))
+        .thenComposeAsync((resp) -> handleRedirects(redirectsLeft - 1, request, resp))
+        .toCompletableFuture();
+    } else {
+      return CompletableFuture.completedFuture(response);
+    }
+  }
+
```

Comment:
Should we move this into Akka itself? even if only working on the high level API it'd be nice anyway to have it there?

cc @jrudolph